src/components/DateRangePicker.js

```diff
@@ -0,0 +1,353 @@
+    }
+    if (this.state.selectedEndDate === date) {
+      this.state.selectedEndDate = null;
+      return;
+    }
+
+    if (!this.state.selectedStartDate || this.state.selectedStartDate > date) {
+      this.state.selectedStartDate = date;
+    } else if (this.state.selectedStartDate < date) {
+      this.state.selectedEndDate = date;
+    }
+
+    this.props.onDateSelect(this.state.selectedStartDate, this.state.selectedEndDate);
+  },
+
+  _handleDateHover (date) {
+    this.state.activeSelectDate = date;
+  },
+
+  _handlePreviousClick () {
+    const currentDate = moment.unix(this.state.currentDate).startOf('month').subtract(1, 'm').unix();
+
+    this.setState({
+      currentDate
+    });
+  },
+
+  _handleNextClick () {
+    const currentDate = moment.unix(this.state.currentDate).endOf('month').add(1, 'd').unix();
+
+    this.setState({
+      currentDate
+    });
+  },
+
+  _handleScrimClick () {
+    this.setState({
+      showCalendar: false
+    });
+  },
+
+  _toggleCalendar () {
+    this.setState({
+      showCalendar: !this.state.showCalendar
+    });
+  },
+
+  _renderMonthTable () {
+    const styles = this.styles();
+    const days = [];
+    let startDate = moment.unix(this.state.currentDate).startOf('month').startOf('week');
+    const endDate = moment.unix(this.state.currentDate).endOf('month').endOf('week');
+
+    while (moment(startDate).isBefore(endDate)) {
+      const isToday = startDate.isSame(moment(), 'day');
+      const isCurrentMonth = startDate.isSame(moment.unix(this.state.currentDate), 'month');
+      const disabledDay = this.props.minimumDate ? startDate.isBefore(moment.unix(this.props.minimumDate)) : null;
+      const isSelectedStartDay = startDate.isSame(moment.unix(this.state.selectedStartDate), 'day');
+      const isSelectedEndDay = startDate.isSame(moment.unix(this.state.selectedEndDate), 'day');
+      const isActiveRange = startDate.isBetween(moment.unix(this.state.selectedStartDate), moment.unix(this.state.activeSelectDate));
+      const isSelectedRange = startDate.isBetween(moment.unix(this.state.selectedStartDate), moment.unix(this.state.selectedEndDate));
+
+      const day = (
+        <div
+          key={startDate}
+          onClick={disabledDay ? null : this._handleDateSelect.bind(null, startDate.unix())}
+          onMouseOver={(disabledDay || !this.state.selectedStartDate || this.state.selectedEndDate) ? null : this._handleDateHover.bind(null, startDate.unix())}
+          style={Object.assign({},
+            styles.calendarDay,
+            isCurrentMonth && styles.currentMonth,
+            disabledDay && styles.calendarDayDisabled,
+            (isToday && (!isActiveRange || !isSelectedRange)) && styles.today,
+            isSelectedStartDay && Object.assign({}, styles.selectedDay, styles.selectedStartDay),
+            isSelectedEndDay && Object.assign({}, styles.selectedDay, styles.selectedEndDay),
+            (isSelectedRange || isActiveRange) && styles.betweenDay
```

Comment:
isActiveRange is being declared before it's set on line 117 which is why the event is one behind the user. However= even if I check to see if the dates are between here, I see the same behavior.